### DISCUSSION TITLE
how to create / read secrets using the library

### DISCUSSION THREAD

**Discussion by vigouredelaruse** [kubernetes-client/csharp#1283] [Q&A] [answered]
i looked through the code=>models and there's no secet object (like [this](https://github.com/Azure/azure-functions-core-tools/blob/v4.x/src/Azure.Functions.Cli/Kubernetes/Models/Kubernetes/SecretsV1.cs))

so, how does one 
* create a secret in a namespace
* read a secret from a namespace

please advise

**Comment by tg123** [comment, marked as answer]
`CreateNamespacedSecret`?

**Reply by vigouredelaruse** [reply]
i can see a useage example as per https://github.com/akanso/complete-cs-k8s-example/blob/master/src/Program.cs so i'm going to mark this as the answer - 

thanks very much
```
            [V1Secret.KubeKind] = async (k, b, ns, ct) => await k.CreateNamespacedSecretAsync((V1Secret)b, ns, cancellationToken: ct),
```
